packages/start/README.md

```diff
@@ -66,6 +66,122 @@ const [theme, setTheme] = createUserTheme("cookieName", {
 theme(); // => "light" | "dark"
 ```
 
+
+## Examples
+
+### `root.tsx`
+
+This is the main entry file for SolidStart demonstrating usage of the `createUserTheme` function.
+- Initializes the theme with the `createUserTheme` function.
+- Toggles between the "dark" and "light" themes using a button.
+- Sets the `data-theme` attribute adding compatibility between `createUserTheme` and libraries like [DaisyUI](https://daisyui.com/).
+- Includes a clean implementation of theming through a `Body` class.
+- Provides additional examples on using variant classes to style components.
+
+```tsx
+// @refresh reload
+import { Suspense } from "solid-js";
+import {
+  A,
+  Body,
+  ErrorBoundary,
+  FileRoutes,
+  Head,
+  Html,
+  Meta,
+  Routes,
+  Scripts,
+  Title,
+} from "solid-start";
+import Counter from "./components/Counter";
+import "./root.css";
+import { createUserTheme } from "./start_primitive/index";
+export default function Root() {
+  const [theme, setTheme] = createUserTheme("dark-light-theme", "dark");
+  const toggleTheme = () => setTheme(currentTheme => currentTheme === "dark" ? "light" : "dark");
+  return (
+    // Set data-theme when using libraries like Daisyui:
+    <Html lang="en" data-theme={theme()}>
+      <Head>
+        <Title>SolidStart - Bare</Title>
+        <Meta charset="utf-8" />
+        <Meta name="viewport" content="width=device-width, initial-scale=1" />
+      </Head>
+      <Body
+        classList={{
+          // Apply theme as css class to body:
+          [theme()]: true,
+        }}
+      >
+        <div>
+          <Counter
+            classList={{
+              // Within components conditionally apply css classes:
+              "dark-variant": theme() !== "light",
+              "light-variant": theme() === "light",
```

Comment:
again, wouldn’t you want to switch it like this?
```tsx
class={theme() === "light" ? "light-variant" : "dark-variant"}
```

Reply:
I could use both methods and comment out `classList` again.

Reply:
Looking at this again, I don't think you'd use the theme at the component level other than special situations and even then you would most likely apply theme dependent variants through `classList`.